txselector/txselector.go

```diff
@@ -1121,6 +1125,94 @@ func filterFailedAtomicGroups(
 	return txsToProcess, filteredTxs
 }
 
+// filterInvalidAtomicGroups split the txs into the ones that can be porcessed
+// and the ones that can't because they belong to an AtomicGroup that is impossible to forge
+// due to missing or bad ordered txs
+func filterInvalidAtomicGroups(
+	txs []common.PoolL2Tx,
+) (txsToProcess []common.PoolL2Tx, filteredTxs []common.PoolL2Tx) {
+	// validate if an atomic group is valid in terms of requested txs
+	isValid := func(atomicGroup common.AtomicGroup) bool {
```

Comment:
I'm not sure about the impact of evaluating this function variable every time this function is called, maybe we could make this as a static private function in the same package with the name isAtomicGroupValid to help to make this filterInvalidAtomicGroups less complex and improve code readability.